### PR TITLE
#5736 Group similar error reports

### LINKED ISSUE #5736: Group similar error reports
There are many error reports which are sent in [doSendErrorToSharedTenantFes](https://github.com/FlowCrypt/flowcrypt-browser/blob/d4c6c45fadccc244e3de726427533c3c4dec6345/extension/js/common/platform/catch.ts#L313) with similar message like `429 when GET-ing https://www.googleapis.com/gmail/v1/users/me/threads/18fa6109b0f0d121 (no body)`, which has the only difference - thread id.

For such cases we should just replace such parameters with `THREAD_ID`, so similar reports will be grouped on our backend.

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented feature to group similar error reports

close #5736 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ioanmo226** [comment]
@sosnovsky Please check

File: extension/js/common/platform/catch.ts [hunk 1]
```diff
@@ -281,6 +281,14 @@ export class Catch {
     }
   }
 
+  private static groupSimilarReports(value: string): string {
+    return value
+      .replace(/chrome-extension:\/\/[^\/]+\//, 'chrome-extension://EXTENSION_ID/')
+      .replace(/https:\/\/www\.googleapis\.com\/gmail\/v1\/users\/me\/threads\/[^\/]+\//, 'https://www.googleapis.com/gmail/v1/users/me/threads/THREAD_ID/')
+      .replace(/https:\/\/www\.googleapis\.com\/gmail\/v1\/users\/me\/messages\/[^\/]+\//, 'https://www.googleapis.com/gmail/v1/users/me/messages/MESSAGE_ID/')
+      .replace(/https:\/\/www\.googleapis\.com\/gmail\/v1\/users\/me\/drafts\/[^\/]+\//, 'https://www.googleapis.com/gmail/v1/users/me/drafts/DRAFT_ID/');
+  }
```
Comment (sosnovsky) [inline extension/js/common/platform/catch.ts:290]: currently only `EXTENSION_ID` replacement works, in other regexes trailing slashes should be removed - as urls look like https://www.googleapis.com/gmail/v1/users/me/messages/12079801d03027dc

Comment (ioanmo226) [reply extension/js/common/platform/catch.ts]: Ohhh. misssed it.
Just fixed it

**Review by sosnovsky — APPROVED** [review]
Works well now, thanks for quick fix!

**Pull request merged**